File: Java Programs/Chapter 1/BinarySearch/src/binarysearch/BinarySearch.java

package binarysearch;

//Program to search an element in an array using binary search.
//Array A must be in sorted order and its maximum size is 20.
import java.util.*;

public class BinarySearch
{
	public static int BinarySearch(int A[],int low,int high,int e)
	{
		
		 int mid;
                 if(low>high)
                 {
                    System.out.println("Element is not found");
                    return -1;
                 }
                mid=(low+high)/2;
                if(A[mid]==e)
                {
                    System.out.println("Element found in Position:"+mid);
                    return mid;
                }
                if(A[mid]>e)
                {
                    return BinarySearch(A,low,mid-1,e);
                }
                else
                {
                    return BinarySearch(A,mid+1,high,e);
                }
               
	}
	public static void main(String args[])
	{
		int[] A=new int[20];
		int i,n,pos,low,high;
		int e;
		Scanner oin=new Scanner(System.in);
		
		//Getting the elements in the array.
		System.out.print("Enter the number of elements in the array: ");
		n=Integer.parseInt(oin.nextLine());
			
		System.out.println("Enter the elements of the array:");
		for(i=1;i<=n;i++)
		{
			A[i]=Integer.parseInt(oin.nextLine());
		}
		
		System.out.println("Enter the element to search:");
		e=Integer.parseInt(oin.nextLine());
		
		low=1;
		high=n;
		pos=BinarySearch(A,low,high,e);
		
	}
}

File: Python Programs/unit1/BinarySearch.py
from __future__ import print_function


def Binary_Search(array, low, high, elementToSearch):
    if low > high:
        print("Element not found")
        return -1
    mid = (low + high) / 2
    if array[mid] == elementToSearch:
        print("Element is found in the array")
        return mid
    if array[mid] > elementToSearch:
        Binary_Search(array, low, mid - 1, elementToSearch)
    else:
        Binary_Search(array, mid + 1, high, elementToSearch)


arr = []
print("Enter the number of elements in an array")
n = int(raw_input())
print("Enter the elements")
for i in range(0, n):
    arr.append(int(raw_input()))
print("Elements in the array")
for i in arr:
    print("%d" % i)
print("Enter the element to search")
e = int(input())
Binary_Search(array=arr, low=0, high=n - 1, elementToSearch=e)
